src/mongo/db/curop.h

```diff
@@ -305,6 +306,10 @@ class CurOp {
         if (_dbprofile <= 0)
             return false;
 
+        if (serverGlobalParams.rateLimit > 1 && _dbprofile >= 2 && ms < serverGlobalParams.slowMS) {
```

Comment:
Event are not limited for that case. Events longer than slowms are not limited when profile = 2. So I decided not limit them when profile = 1 too.